### PR TITLE
Port over some basic overrides/changes from MST Extra, fix tree regrowth

### PR DESCRIPTION
**Pull request by chaosvolt**
#### Summary

SUMMARY: Balance "Port over several useful changes from MST Extra mod, fix tree regrowth"

#### Purpose of change

I decided I'd finally port over some of the easy changes from the BN version of MST Extra. Basically just overrides to various things. This quickly turned into more of a fix for something related to one of the things I intended to port over that ended up being shelved for now.

#### Describe the solution

1. Ported over option to use birchbark as fletching material, as it is a known technique and helps make plastic and paper less of a roadblock to crafting arrows innawoods.
2. On further testing this rapidly turned into me figuring out that the regrowth for a TON of trees were broken, and roughly figuring out what was needed to fix them. The original fix to birchbark that mandated this fix had to be shelved for now though.

#### Describe alternatives you've considered

Some additional overrides and a restoration of craftable stone spears was floated as additional ideas, but went ahead with not porting them over per feedback.

Additionally, discussion led to realizing that one of those changes meant for mod-compatibility stuff is completely unnecessary and can be done differently.

#### Testing

Checked affected files for syntax and lint errors.

Further tests in debug indicated which plants did and did not regrow after a season.

#### Additional context

Birchbark being able to regrow was part of the initial planned port but this got shelved due to it being seen as regrowing too fast. I think I know a way we can fix that but it will warrant a separate PR.

Easy shit I could consider porting over if asked:
1. Jerked offal as an additional way to preserve organs.
2. Leather cordage as a supplement to natural cordage items.
3. Ability to convert an empty window frame into a door frame with an axe or adze.
4. Charcoal water purifier as a reloadable furniture tool.
5. Options to build certain basic constructions like window/door frames using adhesive instead of nails.
6. Option to build wooden stairs with logs and presumed use of joinery instead of nails.
7. Arm and leg wraps as a further compliment to the existing wrap items.
8. Birchbark sheathes since birchbark quivers are a thing.
9. Roasted garlic and chili peppers, garlic bread, cooked cattail rhizomes, and wheat/barley porridge.
10. Side drums as another simple craftable instrument, odd that I don't think there are ANY percussion instruments in the game, far as I remember.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Side note I didn't notice until recently: blackjack oak and pine will regrow, willow and birch don't. So birch wasn't COMPLETELY alone in being the only tree that has no proper regrowth state, but it is a rarity. Can set it so willow trees will regrow too if desired.

**Comment by Coolthulhu** [comment]
I don't see much benefit to 1-4. It will make the items display as ammo and they suck as ammo. There's no good reason to use them as ammo.
6 has same role as wooden spear and is barely different from it. Bloaty in current state.

5 - which actions are enabled by this change? Sounds like there is a chance of weirdness.

7-9 look good.

**Comment by chaosvolt** [comment]
1-4 is mostly a consistency/flavor thing, true. Sulfur is mildly useful as such since it can blind targets. The sports ball items would've been useful as occasional city-specific alternatives in a pinch but those are all obsoleted. Can revert those changes when I get home I guess.

Spear mostly has the advantage of being a modest upgrade from the wooden spear that's more accessible innawoods than the variious makeshift metal spears, and favoring more materials over needing a fire. Can revert that too I guess, unless there's any stat tweaks you feel would make it more worthwhile? Possibly a nudge closer to the makeshift metal spears would make it lean more towards being a reasonable innawoods sidegrade, not sure.

5 doesn't enable any actions, technically. It only comes into play in the event of a recipe that calls for an active lighter being used as a tool, something that's not (to my knowledge) done in vanilla but does have some mod usage. Though now that I think about it, since ember carriers are a vanilla item it should be just as easy to call for their tool use directly in any relevant mod recipes, so I could afford to remove this change entirely.

**Comment by chaosvolt** [comment]
> But then it wouldn't really be a stone spear. I don't see a good way of salvaging the stone spear itself.

Fair enough, I'll revert the change in a bit.

> So what does it change?

It defines the lit ember carrier as a `sub` for a lit refillable lighter. Like I said, I realized the intended interaction can be done entirely at the level of any recipe asking for a lit lighter as a tool, so I'll revert that too.

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 1]
```diff
@@ -934,7 +934,8 @@
     "description": "A tall deciduous tree of the 'Betula' genus, with the characteristic peeling bark.  Species like 'B. alleghaniensis', 'B. papyrifera', and 'B. populifolia' are the most common in the New England region.  Looks like there isn't enough rhytidome to peel off yet.  You could cut it down with the right tools.",
     "symbol": "7",
     "color": "green",
-    "//": "dead, not harvestable",
+    "transforms_into": "t_tree_birch",
```
Comment (Coolthulhu) [inline data/json/furniture_and_terrain/terrain-flora.json:937]: Can't get this to trigger. Same for harvested willow. It might need something more.

Comment (chaosvolt) [reply data/json/furniture_and_terrain/terrain-flora.json:937]: Huh, that's weird. Anything that might be missing from it?

**Comment by chaosvolt** [comment]
So I'm seeing that some harvested trees have the `HARVESTED` flag, some don't. Some also have `transforms_into` but not the `harvested_plant` examine action. Any idea which ones actually work and which don't, and thus which missing element is actually essential to the trees that were edited?

**Comment by chaosvolt** [comment]
Okay, I tested pine trees and blackjack oak, two trees I made no alterations to. Even after skipping two seasons ahead, saving and loading multiple times, I can't get them to regrow either.

Is time skip via debug bugged, or is tree regrowth broken again?

**Comment by chaosvolt** [comment]
Okay, in a follow-up test, the above method works with underbrush, but not pine or blackjack trees.

**Comment by chaosvolt** [comment]
I'm doing more tests, but on adding the `HARVESTED` flag to harvested pine trees, they correctly regrew. Guess this is turning into a general fix PR...

**Comment by Coolthulhu** [comment]
It makes the trees regrow every season. This is too fast for trees that lose their bark.
I mean, it's not a big deal balance-wise, but willow and birch bark is not a balance-related thing at all.

**Comment by chaosvolt** [comment]
Hmm. This did drift a bit due to discovering how broken harvest regrowth was.

One issue however is how most of these trees seem to have been set up as if they were intended to regrow, but didn't. Hence suspecting that this was a bug.

**Pull request merged**